### PR TITLE
[thanos] Add support for service accounts (#930)

### LINKED ISSUE #930: [thanos] add configuration support for pod service accounts
**Is your feature request related to a problem? Please describe.**
Using the default service account for a namespace can be a pain for managing some secrets and external constructs like GCP's [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).

**Describe the solution you'd like to see**
Add a new flag that allows overriding service accounts for the deployments in the Thanos chart.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by seanson**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #930
| License         | Apache 2.0

### What's in this PR?
New helm overrides for settings a service account and README parameters.

### Additional context
Tested on a GKE 1.13.7 cluster with Helm 2.11.0

### Checklist
- [X] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: thanos/README.md [hunk 1]
```diff
@@ -212,6 +213,7 @@ These values are just samples, for more fine-tuning please check the values.yaml
 | query.stores | Addresses of statically configured store API servers (repeatable). The scheme may be prefixed with 'dns+' or 'dnssrv+' to detect store API servers through respective DNS lookups. | [] |
 | query.extraEnv | Add extra environment variables | [] |
 | query.extraArgs |Add extra arguments | [] |
+| store.serviceAccount | Name of the Kubernetes service account to use | "default" |
```
Comment (tarokkk) [inline thanos/README.md]:
```suggestion
| query.serviceAccount | Name of the Kubernetes service account to use | "default" |
```

File: thanos/README.md [hunk 2]
```diff
@@ -227,6 +229,7 @@ These values are just samples, for more fine-tuning please check the values.yaml
 | retentionResolution1h | How long to retain samples of resolution 2 (1 hour) in bucket. 0d - disables this retention | 1y |
 | compactConcurrency | Number of goroutines to use when syncing block metadata from object storage. | 20 |
 | blockSyncConcurrency | Number of goroutines to use when compacting groups. | 1 |
+| store.serviceAccount | Name of the Kubernetes service account to use | "default" |
```
Comment (tarokkk) [inline thanos/README.md]:
```suggestion
| compact.serviceAccount | Name of the Kubernetes service account to use | "default" |
```

File: thanos/README.md [hunk 3]
```diff
@@ -238,6 +241,7 @@ These values are just samples, for more fine-tuning please check the values.yaml
 | bucket.timeout | Timeout to download metadata from remote storage | 5m |
 | bucket.label | Prometheus label to use as timeline title | "" |
 | bucket.http.port | Listening port for bucket web | 8080 |
+| store.serviceAccount | Name of the Kubernetes service account to use | "default" |
```
Comment (tarokkk) [inline thanos/README.md]:
```suggestion
| bucket.serviceAccount | Name of the Kubernetes service account to use | "default" |
```

File: thanos/templates/compact-deployment.yaml [hunk 4]
```diff
@@ -87,4 +87,5 @@ spec:
       {{- with .Values.compact.tolerations }}
       tolerations: {{ toYaml . | nindent 8 }}
       {{- end }}
+      serviceAccountName: "{{ .Values.compact.serviceAccount }}"
```
Comment (tarokkk) [inline thanos/templates/compact-deployment.yaml]:
```suggestion
      {{- with  .Values.compact.serviceAccount }}
      serviceAccountName: "{{ . }}"
      {{- end }}
```

File: thanos/templates/query-deployment.yaml [hunk 5]
```diff
@@ -108,4 +108,5 @@ spec:
       {{- with .Values.compact.tolerations }}
       tolerations: {{ toYaml . | nindent 8 }}
       {{- end }}
+      serviceAccountName: "{{ .Values.query.serviceAccount }}"
```
Comment (tarokkk) [inline thanos/templates/query-deployment.yaml]:
```suggestion
      {{- with  .Values.query.serviceAccount }}
      serviceAccountName: "{{ . }}"
      {{- end }}
```

File: thanos/templates/store-deployment.yaml [hunk 6]
```diff
@@ -109,4 +109,5 @@ spec:
       {{- with .Values.compact.tolerations }}
       tolerations: {{ toYaml . | nindent 8 }}
       {{- end }}
+      serviceAccountName: "{{ .Values.store.serviceAccount }}"
```
Comment (tarokkk) [inline thanos/templates/store-deployment.yaml]:
```suggestion
      {{- with  .Values.store.serviceAccount }}
      serviceAccountName: "{{ . }}"
      {{- end }}
```

File: thanos/values.yaml [hunk 7]
```diff
@@ -127,6 +127,7 @@ store:
   # Pod affinity
   # Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#node-affinity
   affinity: {}
+  serviceAccount: "default"
```
Comment (tarokkk) [inline thanos/values.yaml]:
```suggestion
  // Set service account name for Pod
  serviceAccount: ""
```

File: thanos/values.yaml [hunk 8]
```diff
@@ -195,6 +196,7 @@ query:
       #  - secretName: chart-example-tls
       #    hosts:
       #      - chart-example.local
+  serviceAccount: "default"
```
Comment (tarokkk) [inline thanos/values.yaml]:
```suggestion
  // Set service account name for Pod
  serviceAccount: ""
```

File: thanos/values.yaml [hunk 9]
```diff
@@ -331,6 +333,7 @@ compact:
     # Enable ServiceMonitor https://github.com/coreos/prometheus-operator
     serviceMonitor:
       enabled: false
+  serviceAccount: "default"
```
Comment (tarokkk) [inline thanos/values.yaml]:
```suggestion
  // Set service account name for Pod
  serviceAccount: ""
```

File: thanos/values.yaml [hunk 10]
```diff
@@ -442,6 +445,7 @@ bucket:
   # Pod affinity
   # Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#node-affinity
   affinity: {}
+  serviceAccount: "default"
```
Comment (tarokkk) [inline thanos/values.yaml]:
```suggestion
  // Set service account name for Pod
  serviceAccount: ""
```

**Comment by seanson** [comment]
> I refactored the serviceAccount name settings and fixed some typo. Otherwise OK for me

Thanks for the feedback! I'll get these pushed up.

**Comment by seanson** [comment]
I've re-pushed the changes according to the review.
Also in testing this in our cluster I noticed there are a few greedy `{{- end -}}` invocations that were causing the `---` separator to combine with the ingress ports. I've added this as another commit as it's a small fix but can drop out into another issue if needed.

**Comment by tarokkk** [comment]
I'm sorry to kept you waiting on this PR. However I'm mostly available at https://slack.banzaicloud.io/ in the #banzai-charts channel. Feel free to join and discuss chart related issues

**Pull request merged**